### PR TITLE
Onboarding migration

### PR DESCRIPTION
**Pull request by mstephen19**
Migrating the [onboarding from Gitlab](https://gitlab.com/apify-public/wiki/-/tree/master/wiki/marketplace) into new and revamped lessons within the **pro_apify_scraper_development** folder.

### CODE REVIEW

**Comment by mstephen19** [comment]
Accidentally committed before finishing the message. Wrote **Apify API & client**

**Comment by mstephen19** [comment]
Done making any further large changes to this PR. Will only make changes based on comments.

**Review by mnmkng — APPROVED** [review]
Looks good to me. I haven't read the texts extremely carefully, because I'm sure @metalwarrior665 knows them better.

**Comment by m-murasovs** [comment]
@mstephen19 did you test this using the `?version` thing? Looks like there's something wrong with the structure since it returns a 500 error

**Comment by mstephen19** [comment]
> @mstephen19 did you test this using the `?version` thing? Looks like there's something wrong with the structure since it returns a 500 error

Yeah I thought that was just an issue with the ?version thing. I'll take a look at it. No linting errors being thrown and all the paths seem to be fine, so a bit confused on why this is happening

**Comment by m-murasovs** [comment]
> Yeah I thought that was just an issue with the ?version thing. I'll take a look at it. No linting errors being thrown and all the paths seem to be fine, so a bit confused on why this is happening

The `version` doesn't lie 🙂 but if it stops working in the future, please let the web team know - it's important that it keeps working.

From a look at the structure I see that you have a `solutions` directory but no `solutions.md` file here:

Screenshot 2022-04-18 at 11 24 20

**Comment by mstephen19** [comment]
@m-murasovs Should be fixed now :D

**Comment by mstephen19** [comment]
Accidentally clicked on close

File: content/academy/pro_apify_scraper_development/apify_api_and_client.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    // The maximum number of items to return back
+    "maxItems": 10
+}
+```
+
+[**Solution**]({{@link pro_apify_scraper_development/solutions/using_api_and_client.md}})
+
+## [](#next) Next up
+
+Wow, you've learned a whole lot so far. The next lesson will surely be released soon, as we release new lessons as we write them, so stay tuned!
```
Comment (m-murasovs) [inline content/academy/pro_apify_scraper_development/apify_api_and_client.md:62]: Maybe rephrase to 
```suggestion
Wow, you've learned a whole lot so far. The next lesson will be released as soon as it's written, so stay tuned!
```

File: content/academy/pro_apify_scraper_development/managing_source_code.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+title: Managing source code
+description: Learn how to manage your actor's source code more efficiently by integrating it with a Github repository. This is the standard on the Apify platform.
+menuWeight: 6.3
+paths:
+    - pro-apify-scraper-development/managing-source-code
+---
+
+# [](#managing-source-code) Managing source code
+
+In this brief lesson, we'll discuss how to better manage an actor's source code. Up 'til now, you've been developing your scripts locally, then pushing the code directly to the actor on the Apify platform; however, there is a much more optimal _(and standard)_ way.
```
Comment (m-murasovs) [inline content/academy/pro_apify_scraper_development/managing_source_code.md:11]: No need for italics, just brackets will do
```suggestion
In this brief lesson, we'll discuss how to better manage an actor's source code. Up 'til now, you've been developing your scripts locally, then pushing the code directly to the actor on the Apify platform; however, there is a much more optimal (and standard) way.
```

File: content/academy/pro_apify_scraper_development/managing_source_code.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+## [](#our-task) Our task
+
+> This lesson's task is so quick and easy, we won't even be splitting this topic into two parts like the previous two topics!
+
+First, we must initialize a Github repository (you can use Gitlab if you'd like, but this lesson's examples will be using Github). Then, after pushing our main Amazon actor's code to the repo, we must switch our actor's source code to use the content of the Github repository instead.
+
+## [](#integrating-github) Integrating Github source code
+
+First, let's create a repository. This can be done [in a number of ways](https://kbroman.org/github_tutorial/pages/init.html), but the way we'll do it is by creating the remote repository on Github's website:
```
Comment (m-murasovs) [inline content/academy/pro_apify_scraper_development/managing_source_code.md:35]:
```suggestion
First, let's create a repository. This can be done [in a number of ways](https://kbroman.org/github_tutorial/pages/init.html), but we'll do it by creating the remote repository on Github's website:
```

File: content/academy/pro_apify_scraper_development/managing_source_code.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+So, it was bit of overhead, but the good news is that you don't ever have to configure this stuff again for this actor. Now, every time the content of your **main**/**master** branch changes, the actor on the Apify platform will rebuild based off of the newest code.
+
+A great way to think about it is that, by doing this, you are combining two steps into one! Normally, you'd have to do a `git push` from your terminal in order to get the newest code onto Github, then run `apify push` to push it to the platform.
+
+It's also important to know that Github/Gitlab repository integration is standard practice. As projects grow, and the number of contributors and maintainers grows, it only makes sense to have a Github repository integrated with the project's actor. For the remainder of this course, all actors created will be integrated with a Github repository.
+
+## [](#next) Next up
+
+[Next up]({{@link pro_apify_scraper_development/tasks_and_storage.md}})), you'll learn about the different ways to store scraped data, as well as how to utilize a cool feature to run pre-configured actors.
```
Comment (m-murasovs) [inline content/academy/pro_apify_scraper_development/managing_source_code.md:67]: Cheeky extra bracket
```suggestion
[Next up]({{@link pro_apify_scraper_development/tasks_and_storage.md}}), you'll learn about the different ways to store scraped data, as well as how to utilize a cool feature to run pre-configured actors.
```

File: content/academy/pro_apify_scraper_development/solutions/actor_building.md [hunk 5]
```diff
@@ -18,7 +18,7 @@ apify create demo-actor
 
 Once you run this command, you'll get prompted into a menu which you can navigate using your arrow keys. Each of these options will generate different boilerplate code when selected. We're going to work with CheerioCrawler today, so we'll select that one, then press <p><kbd>Enter</kbd><p>.
```
Comment (m-murasovs) [inline content/academy/pro_apify_scraper_development/solutions/actor_building.md:19]: The `<p>` tag messes up the `<kbd>` tag.

Screenshot 2022-04-20 at 13 18 24

File: content/academy/pro_apify_scraper_development/tasks_and_storage.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+Both of these are very different things; however, they are also tied together in many ways. **Tasks** run actors, actors return data, and data is stored in different types of **Storages**.
+
+## [](#tasks) Tasks
+
+Tasks are a very useful feature which allow us to save pre-configured inputs for actors. This means that rather than configuring the actor every time, or rather than having to save screenshots of various different actor configurations, you can store the configurations right in your Apify account instead, and run the actor at will with them.
+
+## [](#storage) Storage
+
+Storage allows us to save persistent data for further processing. As you'll learn, there are two main storage options on the Apify platform, as well as two main storage types _(**named** and **unnamed**)_ with one big difference between them.
```
Comment (m-murasovs) [inline content/academy/pro_apify_scraper_development/tasks_and_storage.md:19]: No italics here - the brackets are enough
```suggestion
Storage allows us to save persistent data for further processing. As you'll learn, there are two main storage options on the Apify platform, as well as two main storage types (**named** and **unnamed**) with one big difference between them.
```

File: content/academy/pro_apify_scraper_development/tasks_and_storage.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+We have decided that we want to retain the data scraped by the actor for a long period of time, so instead of pushing to the default dataset, we will be pushing to a named dataset. Additionally, we want to save the absolute cheapest item found by the scraper into the default key-value store under a key named **CHEAPEST-ITEM**.
+
+Finally, we'll create a task for the actor that saves the configuration with the **keyword** set to be **google pixel**.
+
+[**Solution**]({{@link pro_apify_scraper_development/solutions/using_storage_creating_tasks.md}})
+
+## [](#next) Next up
+
+The [next lesson]({{@link pro_apify_scraper_development/apify_api_and_client.md}})) is very exciting, as it will unlock the ability to seamlessly integrate your Apify actors into your own external projects and applications with the Apify API.
```
Comment (m-murasovs) [inline content/academy/pro_apify_scraper_development/tasks_and_storage.md:46]: Cheeky extra bracket
```suggestion
The [next lesson]({{@link pro_apify_scraper_development/apify_api_and_client.md}}) is very exciting, as it will unlock the ability to seamlessly integrate your Apify actors into your own external projects and applications with the Apify API.
```

**Review by m-murasovs — CHANGES_REQUESTED** [review]

File: content/academy/pro_apify_scraper_development/solutions/actor_building.md [hunk 8]
```diff
@@ -18,7 +18,7 @@ apify create demo-actor
...
 
-![apify create command]({{@asset pro_apify_scraper_development/images/apify-create.webp}})
+![apify create command]({{@asset pro_apify_scraper_development/solutions/images/apify-create.webp}})
```
[the remark this answers is not in the document]
Comment (mstephen19) [reply content/academy/pro_apify_scraper_development/solutions/actor_building.md:21]: I had this in lowercase because the command is literally `apify create` in all lowercase, but I changed it

File: content/academy/pro_apify_scraper_development/solutions/actor_building.md (hunk 5 above)
Comment (mstephen19) [reply content/academy/pro_apify_scraper_development/solutions/actor_building.md:19]: The `<kbd>` tag doesn't work then. It wasn't working standalone, so I will change all of these tags to bold instead.

File: content/academy/pro_apify_scraper_development/tasks_and_storage.md (hunk 7 above)
Comment (mstephen19) [reply content/academy/pro_apify_scraper_development/tasks_and_storage.md:46]: Life saver!

**Comment by mstephen19** [comment]
Excuse the last commit. Didn't mean for `:wq` to be the message ahaha. There was an error because I forgot to pull latest from the branch before pushing.

File: content/academy/pro_apify_scraper_development/apify_api_and_client.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+title: Apify API & client
+description: Gain an in-depth understanding of the two main ways of programmatically interacting with the Apify platform - throught the API, and through a client.
+menuWeight: 6.5
+paths:
+    - pro-apify-scraper-development/apify-api-and-client
+---
+
+# [](#api-and-client) Apify API & client
+
+There are two main ways to programmatically interact with the Apify platform: By directly using [Apify's RESTful API](https://docs.apify.com/api/v2), or by using the [JavaScript](https://docs.apify.com/apify-client-js) and [Python](https://docs.apify.com/apify-client-python) API clients. In the next two lessons, we'll be focusing on the first two.
```
Comment (m-murasovs) [inline content/academy/pro_apify_scraper_development/apify_api_and_client.md:11]:
```suggestion
There are two main ways to programmatically interact with the Apify platform: by directly using [Apify's RESTful API](https://docs.apify.com/api/v2), or by using the [JavaScript](https://docs.apify.com/apify-client-js) and [Python](https://docs.apify.com/apify-client-python) API clients. In the next two lessons, we'll be focusing on the first two.
```

File: content/academy/pro_apify_scraper_development/apify_api_and_client.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+> Apify's API and JavaScript API client allow us to do anything a regular user can do when interacting with the platform's web interfacee, but programmatically instead.
```
Comment (m-murasovs) [inline content/academy/pro_apify_scraper_development/apify_api_and_client.md:13]:
```suggestion
> Apify's API and JavaScript API client allow us to do anything a regular user can do when interacting with the platform's web interface, only programmatically.
```

**Review by m-murasovs — CHANGES_REQUESTED** [review]
Final lil nitpicks, so after they're fixed, feel free to merge 💪
